lib/core/src/Cardano/Wallet/Primitive/Model.hs

```diff
@@ -139,18 +148,34 @@ data Wallet s t where
         => UTxO -- Unspent tx outputs belonging to this wallet
         -> Set (Tx t, TxMeta) -- Pending outgoing transactions
         -> BlockHeader -- Header of the latest applied block (current tip)
+        -> SlottingState
```

Comment:
Shouldn't this come as a replacement / extension for `BlockHeader` :thinking: ? That's pretty much what the `BlockHeader` used to capture: the wallet slotting state.

Reply:
I'm not sure, but the topic might well be interesting.

My thoughts were that `BlockHeader` should be exactly what was deserialised from the node.

You also know I've previously suggested a `tip` type instead of `BlockHeader`, so I don't view `BlockHeader` as a "tip".